storm-server/src/main/java/org/apache/storm/scheduler/blacklist/BlacklistScheduler.java

```diff
@@ -89,6 +90,7 @@ public void prepare(Map<String, Object> conf) {
         badSupervisorsToleranceSlidingWindow = EvictingQueue.create(windowSize);
         cachedSupervisors = new HashMap<>();
         blacklistHost = new HashSet<>();
+        blacklistOnBadSlots = ObjectReader.getBoolean(this.conf.get(DaemonConfig.BLACKLIST_SUPERVISOR_ON_BAD_SLOT), true);
```

Comment:
We don't need default value here since it's already in `defaults.yaml`

Reply:
Other variables using this format have default value set in default.yaml as well. And the ObjectReader.getBoolean() function require a default input. Should I still make change here?

Reply:
Okay. I am fine with it for now.